### ISSUE TITLE
Webkit Pseudo Class display warning no-unknown

### ISSUE THREAD

**Issue by aviram83** [stylelint/stylelint#3151]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

this is a bug with webkit Pseudo Class

> Which rule, if any, is this issue related to?

selector-pseudo-class-no-unknown

> What CSS is needed to reproduce this issue?

e.g.

```less
      .viewport {
        &::-webkit-scrollbar {
          width: 15px;
          height: 15px;
          background-color: transparent;
        }

        &::-webkit-scrollbar-thumb {
          border-radius: 10px;
          border: 4px solid #fff;
          background-color: #c2c2c2;
          height: 60px;
        }

        &::-webkit-scrollbar-track {
          background-color: #fff;
        }

        &::-webkit-scrollbar-thumb:vertical {
          border: 4px solid rgba(0, 255, 255, 0);
          background-clip: content-box;
        }
```

> What stylelint configuration is needed to reproduce this issue?

e.g.

```json
{
  "extends": "stylelint-config-recommended",
}
```

> Which version of stylelint are you using?

`8.4.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

"npx stylelint "src/**/*.less""

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

"Yes, using ampersand in LESS"

> What did you expect to happen?

"No warnings to be flagged."

> What actually happened (e.g. what warnings or errors you are getting)?

e.g. "The following warnings were flagged:"

```shell
test.less
Unexpected unknown pseudo-class selector ":vertical"   selector-pseudo-class-no-unknown
```

**Comment by hudochenkov** [comment]
Thanks for the report and for using the template.

Rule works as expected. 

> This rule considers pseudo-class selectors defined in the CSS Specifications, up to and including Editor's Drafts, to be known.

There is no `:vertical` pseudo-class: [Index of standard pseudo-classes](https://developer.mozilla.org/en-US/docs/Web/CSS/Pseudo-classes#Index_of_standard_pseudo-classes).

**Comment by aviram83** [comment]
this psedo-class already added by another issue
look at:
https://github.com/stylelint/stylelint/blob/master/lib/reference/keywordSets.js#L263
and
https://github.com/stylelint/stylelint/blob/master/lib/reference/keywordSets.js#L273

so it should work

**Comment by aviram83** [comment]
@hudochenkov also we have `:vertical` in [webkit](https://webkit.org/blog/363/styling-scrollbars/)

**Comment by jeddy3** [comment]
> this pseudo-class already added by another issue

It looks like, at some point, we considered proprietary webkit pseudo elements and classes to been _known_. This seems fair to me.

We have accept tests for them:

https://github.com/stylelint/stylelint/blob/53e8e4b81d3a263584af48258b3f7cefa4cc32f6/lib/rules/selector-pseudo-class-no-unknown/__tests__/index.js#L94-L97

However, in `selector-pseudo-class-no-unknown` we assume these proprietary pseudo elements and classes will be standalone:

https://github.com/stylelint/stylelint/blob/53e8e4b81d3a263584af48258b3f7cefa4cc32f6/lib/rules/selector-pseudo-class-no-unknown/index.js#L92-L95

For example, the following produces no violations:

```css
::-webkit-scrollbar-thumb:vertical {}
```

Whereas the following produces one violation:

```css
.viewport::-webkit-scrollbar-thumb:vertical {}
```

If the second example is valid code, then I think there's a bug.

@aviram83 Please consider [contributing a fix](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#fixing-a-bug-in-an-existing-rule) if you're confident these proprietary webkit pseudo elements and classes can be applied to a simple selector like `.viewport`.

**Comment by rexboy77** [comment]
Seems non-standalone pseudo-class is allowed too. I tried to create [a test page](https://jsfiddle.net/bk96pxyx/4/) and you can see the vertical scrollbar's color changes.

There are test cases contained in Webkit's tests: (https://github.com/WebKit/webkit/blob/master/LayoutTests/scrollbars/overflow-scrollbar-combinations.html#L324)
